rest/security-rest-cxfwrapper/src/test/java/org/codice/ddf/cxf/SecureCxfClientFactoryTest.java

```diff
@@ -0,0 +1,186 @@
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.MessageBodyReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SecureCxfClientFactoryTest {
+
+    private String insecureEndpoint = "http://some.url.com/query";
+
+    private String secureEndpoint = "https://some.url.com/query";
+
+    @Test
+    public void testInsecure() throws SecurityServiceException {
+        // positive case
+        SecureCxfClientFactory<IDummy> secureCxfClientFactory = new SecureCxfClientFactory<>(
+                insecureEndpoint, IDummy.class);
+        Client unsecuredClient = WebClient.client(secureCxfClientFactory.getUnsecuredClient());
+        assertTrue(unsecuredClient.getBaseURI().toASCIIString().equals(insecureEndpoint));
+        // negative cases
+        boolean subject = false;
+        try {
+            secureCxfClientFactory.getClientForSubject(getSubject());
+        } catch (SecurityServiceException e) {
+            subject = true;
+        }
+        assertTrue(subject);
+        boolean system = false;
+        try {
+            secureCxfClientFactory.getClientForBasicAuth(null, null);
+        } catch (SecurityServiceException e) {
+            system = true;
+        }
+        assertTrue(system);
+    }
+
+    @Test
+    public void testSecure() throws SecurityServiceException {
+        // positive cases
+        MockWrapper<IDummy> mockWrapper = new MockWrapper<>(secureEndpoint, IDummy.class);
+        validateConfig(mockWrapper, null, null, false);
+        mockWrapper = new MockWrapper<>(secureEndpoint, IDummy.class);
+        validateConfig(mockWrapper, "foobar", "foobaz", false);
+        List<MockProvider> providers = Arrays.asList(new MockProvider());
+        mockWrapper = new MockWrapper<>(secureEndpoint, IDummy.class, providers, null, true);
+        validateConfig(mockWrapper, "bazfoo", "bazbar", true);
+        // negative case
+        boolean unsecured = false;
+        try {
+            mockWrapper.getUnsecuredClient();
+        } catch (SecurityServiceException e) {
+            unsecured = true;
+        }
+        assertTrue(unsecured);
+    }
+
+    private void validateConfig(MockWrapper<IDummy> factory, String username, String password,
+            boolean disableCnCheck) throws SecurityServiceException {
+        IDummy clientForSubject = factory.getClientForBasicAuth(username, password);
+        HTTPConduit httpConduit = WebClient.getConfig(WebClient.client(clientForSubject))
+                .getHttpConduit();
+        AuthorizationPolicy authorization = httpConduit.getAuthorization();
+        assertTrue(StringUtils.equals(authorization.getUserName(), (username)));
```

Comment:
Unless there's an underlying reason I'm not seeing, it is a bit cleaner to use assertEquals here as it would have better output if an error occurs (example: expected srogers but got bbanner).